ci/gitlab-ci/README.md

```diff
@@ -0,0 +1,13 @@
+<p align="center">
+  <a href="https://secrethub.io/docs/guides/gitlabci/"><img alt="View Docs" src="https://secrethub.io/img/buttons/github/view-docs.png?v2" height="28" /></a>
+</p>
+<br/>
+
+<h1>
+  GitLab CI Examples
+</h1>
+
+- [Publish Docker](./publish-docker/)
+     - [Flavor 1](./publish-docker/flavor-1/.gitlab-ci.yml)
+     - [Flavor 2](./publish-docker/flavor-2/.gitlab-ci.yml)
```

Comment:
Just seeing this pass by, please describe the two use cases. Just `flavor 1` and `flavor 2`  doesn't tell the reader anything about it. Can be just in one sentence.

Reply:
This is also an opportunity to explain when to use which flavor.

Reply:
I am not sure what the advantage of `flavor 1` (using `secrethub read`) is. It does not provide masking and it also looks more complex than the other one. @SimonBarendse Could please tell me in what scenario would this flavor be recommended?

Reply:
I think it's the simplicity of not having an environment variable in between. So you no longer have to create a mapping with an environment variable and inject that. It's a oneliner instead of depending on setting the environment.

However, I think that while the second flavor needs a tiny bit extra code, this actually adds to the readability, as you now have a clearer overview of what secrets are used (in the environment).

Therefore, I think I'd prefer the flavor 2 in both cases.

@florisvdg Do you have anything to add to that?

Reply:
Shall I drop flavor 1, then?

Reply:
Sounds like a plan :+1:

Reply:
Done